### PR TITLE
Fluff Item Port

### PR DESCRIPTION
**Pull request by JoanRisu**
Initial Fluff Item Port! This is mostly for placeholders, debugging, and testing.

Some old code referring to empty magazine clips for guns, FNaF cold protection flags & fingerprint flags, and m_amt were removed to allow for compiling as they were conflicting with new code. 

Items with missing parent file paths were also removed until the missing parent items are added back in. 

I do not have time to do in-depth debugging and continuous testing due to college. I did as much as I could with my small skill set for now.

### CODE REVIEW

**Comment by OrbisAnima** [comment]
I will wait for this one to be pulled to add my custom files, we have to be careful with the dmi files since we can't do an appropiate merge of them.

**Comment by JoanRisu** [comment]
I made sure to sync before editing dmi files but yes we should be careful. I'm gonna try to review at least the dmi before doing anything else. From the looks of Ace's latest merge, he already ported some custom items.

**Comment by OrbisAnima** [comment]
Roger, give me the Go light for them and I will get the files, from what I saw Scree pompom, Tassalds luger and Richard's chain are not yet on, and I might as well get into code Keegan's succubi wings too

**Comment by JoanRisu** [comment]
Should be alright, I didn't notice anything wrong with the dmi files. The only changes from the last merge that compare to mine were to custom_items.dmi

**Comment by JoanRisu** [comment]
So quick update. All energy based fluff guns are confirmed to be in working order.

The only visible problem with all projectile guns is that they cannot be
reloaded. Setting the load_method returns an undefined var error during
compiling.

**Comment by OrbisAnima** [comment]
Roger, I will download your branch and see what's up

**Comment by OrbisAnima** [comment]
Almost sure is that you need to add the /gun/ after energy again. I used the following for Tassald's lugger and it worked like a charm

/obj/item/weapon/gun/energy/gun/fluff/tluger

and I see you using for Dominator 

/obj/item/weapon/gun/energy/fluff/dominator

**Comment by OrbisAnima** [comment]
Add load methods to the weapons that use it, I just tested with

load_method = SINGLE_CASING

to the rifle and it worked like a charm. Now, you can either set it as 1 instead of SINGLE_CASING
or place the following at the top of item_defines.dm

//Gun loading types
#define SINGLE_CASING 	1	//The gun only accepts ammo_casings. ammo_magazines should never have this as their mag_type.
#define SPEEDLOADER 	2	//Transfers casings from the mag to the gun when used. 
#define MAGAZINE 		4	//The magazine item itself goes inside the gun

**Comment by JoanRisu** [comment]
I know about the gun loading types but it kept giving me errors whenever I set it to magazine.

**Comment by JoanRisu** [comment]
Alrighty! All the guns are tested and should work now~ 

There are several things that you have to set for guns and ammo
For ammo, you have to set their mag_type to one of the two loading types (two because SINGLE_CASING should never be a mag_type.
For guns themselves, you have to set the load_method to any of the three gun loading types and set the ammo_type and magazine_type
They must match methods and caliber in order to work.

Basically, there are a bunch of flags that need to match up and it hurts my head.

**Comment by OrbisAnima** [comment]
You did good. now point Ace to this so he can pull it and I can make my own branch with stuff :P

**Comment by JoanRisu** [comment]
Fixed some revolvers reported not working and added guns that were part
of the Nazi event.

**Comment by SpadesNeil** [comment]
So is that everything, then?

**Comment by SpadesNeil** [comment]
Joan, your config for your compiler is set wrong.

Build -> Preferences -> Uncheck automatically set FILE_DIR

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 1]
```diff
@@ -26,7 +28,7 @@
 		recentpump = world.time
 
 /obj/item/weapon/gun/projectile/shotgun/pump/proc/pump(mob/M as mob)
-	playsound(M, 'sound/weapons/shotgunpump.ogg', 60, 1)
+	playsound(M, cocksound /*'sound/weapons/shotgunpump.ogg'*/, 60, 1)
```
Comment (ShadowLarkens) [inline code/modules/projectiles/guns/projectile/shotgun.dm]: commented-out code in the middle of a call is really messy

File: code/modules/customitems/item_defines.dm [hunk 2]
```diff
@@ -1455,3 +1455,1113 @@
...
+	name = "blueshield tactical headset"
+	desc = "A Tactical Radio Headset. It's basically a security headset with a camera. To access the security channel, use :s. For command, use :c."
+	keyslot2 = new /obj/item/device/encryptionkey/heads/hos
+
+// wankersonofjerkin:Ryan Winz
+/obj/item/weapon/storage/briefcase/fluff/ryan_winz
+	name = "Ryan's Gun Case"
+	desc = "Used to safely transport Ryan's two personal revolvers."
+	storage_slots = 2
+	New()
```
Comment (ShadowLarkens) [inline code/modules/customitems/item_defines.dm]: Not especially important, but all of these relative-pathed procs could really do with being absolutely-pathed
instead of
```
/obj/item/weapon/storage/briefcase/fluff/ryan_winz
	name = "Ryan's Gun Case"
	desc = "Used to safely transport Ryan's two personal revolvers."
	storage_slots = 2
	New()
		new /obj/item/weapon/gun/projectile/revolver/mateba/fluff/ryan_winz_revolver(src)
		new /obj/item/weapon/gun/projectile/revolver/mateba/fluff/ryan_winz_revolver/redemption(src)
		..()
		return
```
it would be
```
/obj/item/weapon/storage/briefcase/fluff/ryan_winz
	name = "Ryan's Gun Case"
	desc = "Used to safely transport Ryan's two personal revolvers."
	storage_slots = 2
	
/obj/item/weapon/storage/briefcase/fluff/ryan_winz/New()
	new /obj/item/weapon/gun/projectile/revolver/mateba/fluff/ryan_winz_revolver(src)
	new /obj/item/weapon/gun/projectile/revolver/mateba/fluff/ryan_winz_revolver/redemption(src)
	..()

Comment (SpadesNeil) [inline code/modules/customitems/item_defines.dm]: What's the benefit of this? I never understood.

Comment (ShadowLarkens) [inline code/modules/customitems/item_defines.dm]: Neatness, mostly. It also reduces any possible confusion over what belongs to what. Little things like New() being relative pathed, _right under it_ is generally in the 'okay' territory, but it's generally nicer to have it consistently absolutely pathed.

**Comment by JoanRisu** [comment]
Ok, did some of the suggested changes and set my compiler as requested.
